resource_customizations/ceph.rook.io/CephCluster/health.lua

```diff
@@ -0,0 +1,66 @@
+-- CRD documentation: https://rook.github.io/docs/rook/latest-release/CRDs/Cluster/ceph-cluster-crd/
+local hs = {
+    status = "Progressing",
+    message = ""
+}
+
+function append_to_message(message)
+    if message ~= "" then
+        if hs.message ~= "" then
+            hs.message = hs.message .. " - " .. message
+        else
+            hs.message = message
+        end
+    end
+end
+
+if obj.status == nil then
+    append_to_message("Waiting for status to be reported")
+    return hs
+end
+
+-- Check the main Ceph health status first - https://github.com/ceph/ceph/blob/v20.3.0/src/include/health.h#L12
+if obj.status.ceph ~= nil and obj.status.ceph.health ~= nil then
+    local ceph_health = obj.status.ceph.health
+    local details_message = ""
+
+    -- Build details message from status.ceph.details if available
+    if obj.status.ceph.details ~= nil then
+        local detail_parts = {}
+        for detail_type, detail_info in pairs(obj.status.ceph.details) do
+            if detail_info.message ~= nil then
+                table.insert(detail_parts, detail_info.message)
+            end
+        end
+        if #detail_parts > 0 then
+            details_message =  " (" .. table.concat(detail_parts, "; ") .. ")"
+        end
+    end
+
+    if ceph_health == "HEALTH_ERR" or ceph_health == "HEALTH_WARN" then
+        hs.status = "Degraded"
+    elseif ceph_health == "HEALTH_OK" then
+        hs.status = "Healthy"
+    end
+    append_to_message("Ceph health is " .. ceph_health .. details_message)
+end
+
+-- Check state - https://github.com/rook/rook/blob/v1.17.7/pkg/apis/ceph.rook.io/v1/types.go#L621
+if obj.status.state ~= nil then
+    if hs.status == "Healthy" or hs.status == "Connected" then
```

Comment:
```suggestion
    if hs.status == "Healthy" then
```

I think this was added by mistake. `"Connected"` isn't a value that's set above, and it's not a valid Argo CD health state.

Reply:
Comment addressed in [007b9c3](https://github.com/argoproj/argo-cd/pull/24111/commits/007b9c33764d74889abe258f19d2f1f764fc08cd). Thank you for catching this!